javaharness/java/arcs/android/impl/RuntimeSettingsAndroidJsImpl.java

```diff
@@ -0,0 +1,90 @@
+package arcs.android.impl;
+
+import arcs.api.RuntimeSettings;
+import com.google.auto.value.AutoValue;
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.logging.Logger;
+import javax.inject.Inject;
+
+/** For Javascript-based Arcs runtime. */
+public final class RuntimeSettingsAndroidJsImpl implements RuntimeSettings {
+  // Equivalent to &log parameter
+  private static final String LOG_LEVEL_PROPERTY = "debug.arcs.runtime.log";
+  // Equivalent to &explore-proxy parameter
+  private static final String USE_DEV_SEVER_PROXY = "debug.arcs.runtime.use_alds";
+  // Connects to the host via http://localhost:8786/shells/pipes-shell/...
+  private static final String USE_HOST_PIPES_SHELL = "debug.arcs.runtime.use_host_shell";
+
+  // Default settings: the most logging messages and use Arcs Local Development Server (ALDS)
+  private static final int DEFAULT_LOG_LEVEL = 2;
+  private static final boolean DEFAULT_USE_DEV_SERVER = true;
+  private static final boolean DEFAULT_USE_HOST_SHELL = false;
+
+  private static final Logger logger = Logger.getLogger(
+      RuntimeSettingsAndroidJsImpl.class.getName());
+
+  @AutoValue
+  abstract static class Settings {
+    abstract int logLevel();
+    abstract boolean useDevServerProxy();
+    abstract boolean useHostPipesShell();
+
+    static Builder builder() {
+      return new AutoValue_RuntimeSettingsAndroidJsImpl_Settings.Builder();
+    }
+
+    @AutoValue.Builder
+    abstract static class Builder {
+      abstract Builder setLogLevel(int level);
+      abstract Builder setUseDevServerProxy(boolean useDevServerProxy);
+      abstract Builder setUseHostPipesShell(boolean useHostPipesShell);
+      abstract Settings build();
+    }
+  }
+
+  // Immutable configurations for this runtime settings instance.
+  private final Settings settings;
+
+  @Inject
+  public RuntimeSettingsAndroidJsImpl() {
+    settings = Settings.builder()
+        .setLogLevel(
+            getProperty(LOG_LEVEL_PROPERTY, Integer::valueOf, DEFAULT_LOG_LEVEL))
+        .setUseDevServerProxy(
+            getProperty(USE_DEV_SEVER_PROXY, Boolean::valueOf, DEFAULT_USE_DEV_SERVER))
+        .setUseHostPipesShell(
+            getProperty(USE_HOST_PIPES_SHELL, Boolean::valueOf, DEFAULT_USE_HOST_SHELL))
+        .build();
+  }
+
+  @Override
+  public int logLevel() {
+    return settings.logLevel();
+  }
+
+  @Override
+  public boolean useDevServerProxy() {
+    return settings.useDevServerProxy();
+  }
+
+  @Override
+  public boolean useHostPipesShell() {
+    return settings.useHostPipesShell();
+  }
+
+  private <T> T getProperty(String property, Function<String, T> converter, T defaultValue) {
```

Comment:
Can you add a comment explaining what this section does?

Reply:
Sure, will add some comment and defs of parameter and return values.

Reply:
Done.